src/main/python/deep_qa/layers/noisy_or.py

```diff
@@ -0,0 +1,88 @@
+from keras import backend as K
+from keras.layers import Layer
+from keras.initializations import identity
+from keras.constraints import Constraint, NonNeg
+from keras.regularizers import L1L2Regularizer
+from overrides import overrides
+
+class BetweenZeroAndOne(Constraint):
+    """Constrains the weights to be between zero and one
+    """
+    def __call__(self, p):
+        # handle negatives
+        p *= K.cast(p >= 0., K.floatx())
+        leaky_zeros_mask = K.epsilon() * K.cast(K.equal(p, K.zeros_like(p)), 'float32')
+        p = p + leaky_zeros_mask
+        # handle greater than 1
+        p *= K.cast(p <= 1., K.floatx())
+        leaky_ones_mask = 1.0 * K.cast(K.equal(p, K.zeros_like(p)), K.floatx())
+        p += leaky_ones_mask
+        return p
+
+
+class NoisyOr(Layer):
+
+    def __init__(self, axis=-1, param_init='uniform', prefix="noisy_or", **kwargs):
+        self.axis = axis
+        self.param_init = param_init
+        self.supports_masking = True
+        self.prefix = prefix
+        super(NoisyOr, self).__init__(**kwargs)
+
+    def build(self, input_shape):
+        # Create a trainable weight variable for the noise parameter
+        self.noise_parameter = self.add_weight(shape=(1,),
```

Comment:
pylint is complaining about this being defined outside of `__init__`.  I agree with pylint here - any class variable should show up in `__init__` - it helps with code readability, and IDE tooling.  So just add `self.noise_parameter = None` to the `__init__` method, ideally with a comment about what it's used for (even if it seems obvious).